core/src/main/scala/org/apache/spark/sql/execution/CompactExecRowToMutableRow.scala

```diff
@@ -224,3 +216,20 @@ trait CompactExecRowToMutableRow extends ResultWasNull {
     mutableRow
   }
 }
+
+class ResultNullHolder extends ResultWasNull {
+
+  final var wasNull: Boolean = _
+
+  final val defaultCal = new GregorianCalendar()
```

Comment:
More than a review it will be a doubt clearing exercise :-) ... You can answer in brief?
Why this defaultCal object required in this class?